slider.js

```diff
@@ -0,0 +1,121 @@
+const SLIDER_CLASSES = {
+  LIST: ".slider-list",
+  ITEM: ".slider-item",
+  ITEM_ACTIVE: "slider-item-active",
+  PLACEHOLDER: ".placeholder",
+  ADD_SHOW_PLACEHOLDER: "show-placeholder",
+  NAVIGATION_BUTTONS: ".navigation-buttons",
+  ADD_NAVIGATION_BUTTONS: "showNavigationButtons",
+  BTN_PREV: ".btn-prev",
+  BTN_NEXT: ".btn-next",
+};
+
+const createSlider = (function () {
+  const getSlides = (sliderRoot) => {
+    const list = sliderRoot.querySelector(SLIDER_CLASSES.LIST);
+    const slides = list.querySelectorAll(SLIDER_CLASSES.ITEM);
+    return slides;
+  };
+
+  const getPlaceholder = (sliderRoot) => {
+    sliderRoot
+      .querySelector(SLIDER_CLASSES.PLACEHOLDER)
+      .classList.add(SLIDER_CLASSES.ADD_SHOW_PLACEHOLDER);
+  };
+
+  const getButtons = (sliderRoot) => {
+    const navigation = sliderRoot.querySelector(
+      SLIDER_CLASSES.NAVIGATION_BUTTONS
+    );
+    const prevButton = navigation.querySelector(SLIDER_CLASSES.BTN_PREV);
+    const nextButton = navigation.querySelector(SLIDER_CLASSES.BTN_NEXT);
+
+    return {
+      prevButton,
+      nextButton,
+    };
+  };
+
+  const getPrevSlideIndex = (activeSlideIndex, slides) => {
+    return activeSlideIndex === 0 ? slides.length - 1 : activeSlideIndex - 1;
+  };
+
+  const getNextSlideIndex = (activeSlideIndex, slides) => {
+    return activeSlideIndex >= slides.length - 1 ? 0 : activeSlideIndex + 1;
+  };
+
+  const getPrevSlide = (slides, activeSlideIndex) => {
```

Comment:
getPrevSlide and getNextSlide has a similar logic. What do you think about replacing them with a method selectSlide that will be. selecting any index that will be passed inside?

Reply:
Also there is a responsibility mix. Currently this function modifies active slide and returns new index. It's better either modify active slide or return new index.